src/libServerV2/Server.cpp

```diff
@@ -0,0 +1,1216 @@
+std::mutex Server::m_mutexRecentTxns;
+
+const unsigned int PAGE_SIZE = 10;
+const unsigned int NUM_PAGES_CACHE = 2;
+const unsigned int TXN_PAGE_SIZE = 100;
+
+//[warning] do not make this constant too big as it loops over blockchain
+const unsigned int REF_BLOCK_DIFF = 5;
+
+// Forward declarations (implementation in libMessage).
+void ProtobufToTransaction(const ProtoTransaction& protoTransaction, Transaction& transaction);
+void TransactionToProtobuf(const Transaction& transaction, ProtoTransaction& protoTransaction);
+void ProtobufToDSBlock(const ProtoDSBlock& protoDSBlock, DSBlock& dsBlock);
+void DSBlockToProtobuf(const DSBlock& dsBlock, ProtoDSBlock& protoDSBlock);
+void TxBlockToProtobuf(const TxBlock& txBlock, ProtoTxBlock& protoTxBlock);
+void NumberToProtobufByteArray(const boost::multiprecision::uint256_t& number, ByteArray& byteArray);
+
+
+Server::Server(Mediator& mediator) : m_mediator(mediator) {
+  m_StartTimeTx = 0;
+  m_StartTimeDs = 0;
+  m_DSBlockCache.first = 0;
+  m_DSBlockCache.second.resize(NUM_PAGES_CACHE * PAGE_SIZE);
+  m_TxBlockCache.first = 0;
+  m_TxBlockCache.second.resize(NUM_PAGES_CACHE * PAGE_SIZE);
+  m_RecentTransactions.resize(TXN_PAGE_SIZE);
+  m_TxBlockCountSumPair.first = 0;
+  m_TxBlockCountSumPair.second = 0;
+}
+
+Server::~Server() {
+  // destructor
+}
+
+
+DefaultResponse Server::GetClientVersion() {
+  DefaultResponse ret;
+  return ret;
+}
+
+
+DefaultResponse Server::GetNetworkId() {
+  DefaultResponse ret;
+  ret.set_result("TestNet");
+  return ret;
+}
+
+
+DefaultResponse Server::GetProtocolVersion() {
+  DefaultResponse ret;
+  return ret;
+}
+
+
+DefaultResponse Server::GetGasPrice() {
+  DefaultResponse ret;
+  return ret;
+}
+
+
+DefaultResponse Server::GetStorageAt([[gnu::unused]] GetStorageAtRequest& request) {
+  DefaultResponse ret;
+  return ret;
+}
+
+
+DefaultResponse Server::GetBlockTransactionCount([[gnu::unused]] GetBlockTransactionCountRequest& request) {
+  DefaultResponse ret;
+  return ret;
+}
+
+
+DefaultResponse Server::GetTransactionReceipt([[gnu::unused]] GetTransactionRequest& request) {
+  DefaultResponse ret;
+  return ret;
+}
+
+
+DefaultResponse Server::isNodeSyncing() {
+  DefaultResponse ret;
+  return ret;
+}
+
+
+DefaultResponse Server::isNodeMining() {
+  DefaultResponse ret;
+  return ret;
+}
+
+DefaultResponse Server::GetHashrate() {
+  DefaultResponse ret;
+  return ret;
+}
+
+
+
+////////////////////////////////////////////////////////////////////////////////////////
+
+
+
+CreateTransactionResponse Server::CreateTransaction(CreateTransactionRequest& request) {
+  LOG_MARKER();
+
+  CreateTransactionResponse ret;
+
+  try {
+
+    // Convert Protobuf transaction to Transaction.
+    Transaction tx;
+    if (!request.has_tx()) {
+      ret.set_error("Tx not present in request");
+      return ret;
+    }
+    ProtobufToTransaction(request.tx(), tx);
+
+    // NOTE: Do we need to check if ProtobufToTransaction() failed?
```

Comment:
Any idea how to check whether that failed?

On a side note, there is a verifyTransaction() below. If the protobuf conversion failed, it would be caught here right?